backend/datasource/serve.go

```diff
@@ -0,0 +1,25 @@
+package datasource
+
+import (
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/log"
+)
+
+type Plugin interface {
+	CheckDataSourceHealthHandler
+	CallDataSourceResourceHandler
+	backend.QueryDataHandler
```

Comment:
To be correct, this one should probably have a new interface in this package as well to only include DataSourceConfig in request and not PluginConfig.

Reply:
Could be an interface similar to https://github.com/grafana/grafana/blob/master/pkg/tsdb/query_endpoint.go#L11, basically using similar `DataSource` model struct.